bucket-policy-parser.go

```diff
@@ -264,6 +270,28 @@ func checkBucketPolicyResources(bucket string, bucketPolicy BucketPolicy) APIErr
 	return ErrNone
 }
 
+
+// removeNonePseudoStatements - removes the pseudo 'none' statements (s3:None and s3:NoneBucket) from bucket policy
+func removePseudoNoneStatements(bucketPolicy BucketPolicy) (statements []policyStatement, changed bool) {
+	// Swap all of the none statements to the beginning and return slice of the remaining portion
+	swapIndex := 0
+	for index, statement := range bucketPolicy.Statements {
+		for _, action := range statement.Actions {
+			if strings.HasPrefix(action, "s3:None") {
```

Comment:
It isn't a common prefix, I was just using it so the server can check for any nesting conflicts when we are trying to set 'none' policy. Right after we check for nesting https://github.com/minio/minio/pull/1875/files#diff-4801f881a6fb2336ea40167d248cab0cR208, the none statements are removed https://github.com/minio/minio/pull/1875/files#diff-4801f881a6fb2336ea40167d248cab0cR214.